### ISSUE TITLE
Problems involving the applet after compiling in Manjaro XFCE

### ISSUE THREAD

**Issue by darcn181** [rilian-la-te/vala-panel-appmenu#138]
It seems that in xfce-gtk3(latest version from manjaro repos) is not able to run this applet since it will crash as soon as its added to the panel.

**Comment by rilian-la-te** [comment]
Backtrace and logs, please.

**Comment by darcn181** [comment]
OK, it seems that i had a pretty old version around that was causing the new version to crash, i found it, deleted it and re-compiled the applet, it now works but as you can see it makes some serious changes to user configuration like it forces the use of RGB and changes the icons in use and i can't really change them.
![captura de pantalla_2017-09-24_15-59-13](https://user-images.githubusercontent.com/28988467/30786532-82c52f54-a145-11e7-92d7-4a29b460fb48.png)
![captura de pantalla_2017-09-24_16-19-58](https://user-images.githubusercontent.com/28988467/30786534-82cd63c2-a145-11e7-88f7-87c97ad94ad6.png)
![captura de pantalla_2017-09-24_16-23-38](https://user-images.githubusercontent.com/28988467/30786533-82cc74c6-a145-11e7-886f-81efde6c01ab.png)
Here is the xsession-errors file:
```
`xrdb:  "Xft.hinting" on line 14 overrides entry on line 5
xrdb:  "Xft.hintstyle" on line 17 overrides entry on line 6
gpg-agent[756]: WARNING: "--write-env-file" is an obsolete option - it has no effect
gpg-agent: ya hay un agente gpg ejecutándose - no se inicia otro

(xfce4-session:714): xfce4-session-WARNING **: gpg-agent returned no PID in the variables

(xfce4-session:714): xfce4-session-WARNING **: xfsm_manager_load_session: Something wrong with /home/potato/.cache/sessions/xfce4-session-potato-pc:0, Does it exist? Permissions issue?

(xfce4-session:714): xfce4-session-WARNING **: Unable to launch "xscreensaver -no-splash" (specified by autostart/Xscreensaver.desktop): Falló al ejecutar el proceso hijo «xscreensaver» (No existe el fichero o el directorio)
xfce4-clipman: error while loading shared libraries: libxfconf-0.so.2: cannot open shared object file: No such file or directory
xfce4-panel: No window manager registered on screen 0. To start the panel without this check, run with --disable-wm-check.
xfsettingsd: No window manager registered on screen 0.

(xfsettingsd:772): xfsettingsd-WARNING **: Failed to get the _NET_NUMBER_OF_DESKTOPS property.

(wrapper-2.0:909): Gtk-WARNING **: Negative content width -1 (allocation 1, extents 1x1) while allocating gadget (node button, owner GtkToggleButton)
No window manager registered on screen 0. To start the xfdesktop without this check, run with --disable-wm-check.

(nm-applet:782): Gdk-CRITICAL **: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed

(pa-applet:785): Gdk-CRITICAL **: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed

(pamac-tray:798): Gdk-CRITICAL **: gdk_window_thaw_toplevel_updates: assertion 'window->update_and_descendants_freeze_count > 0' failed

** (xfdesktop:770): WARNING **: Thumbnailer failed calling GetFlavors
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
ThunarThumbnailer: Failed to retrieve supported types: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.thumbnails.Thumbnailer1 was not provided by any .service files
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

(xfce4-panel:766): Gtk-WARNING **: Allocating size to PanelItemDialog 0x55800680e2f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(xfce4-panel:766): Gtk-WARNING **: Allocating size to PanelItemDialog 0x55800680e2f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(xfce4-panel:766): Gtk-WARNING **: Allocating size to PanelItemDialog 0x55800680e2f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?

(xfce4-panel:766): Gtk-WARNING **: Allocating size to PanelItemDialog 0x55800680e2f0 without calling gtk_widget_get_preferred_width/height(). How does the code know the size to allocate?
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

** (wrapper-2.0:1348): WARNING **: Failed to fetch xid: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

** (wrapper-2.0:1348): WARNING **: Failed to fetch type: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

** (wrapper-2.0:1348): WARNING **: Failed to fetch monitor: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

** (wrapper-2.0:1348): WARNING **: Failed to fetch maximized state: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

** (wrapper-2.0:1348): WARNING **: Failed to fetch xid: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

** (wrapper-2.0:1348): WARNING **: Failed to fetch type: GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No existe la interfaz «org.ayatana.bamf.window» en el objeto en la ruta /org/ayatana/bamf/window/12591998

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed
ThunarThumbnailer: Failed to retrieve supported types: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.thumbnails.Thumbnailer1 was not provided by any .service files

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed
ThunarThumbnailer: Failed to retrieve supported types: GDBus.Error:org.freedesktop.DBus.Error.ServiceUnknown: The name org.freedesktop.thumbnails.Thumbnailer1 was not provided by any .service files

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

(wrapper-2.0:1348): Gtk-CRITICAL **: gtk_label_set_text_with_mnemonic: assertion 'str != NULL' failed

````

**Comment by rilian-la-te** [comment]
Does not see anything about RGB or icons in logs:( Only correct warnings, because some program give an empty string to menu.

**Comment by darcn181** [comment]
Ok, this is weird but when the problem happened i rebooted my pc like 3 times, now when i wake up and turn on my pc the problem is solved, even the RGB one and the icons problem, so i'm going to close this issue and if the problem comes back or i find another bug, i'll just reopen this.

**Comment by darcn181** [comment]
The problem came back this is how i have my setup right now, it seems that it changes some icons randomly at startup, i re-logged to my account different time and it changes the icons randomly like, in one boot up it doesnt change the pamac icon and in another it does, i don't know if this is an issue from the part of manjaro or from xfce or from you so i'll try to uninstall it manually and then see if the problem persists

**Comment by rilian-la-te** [comment]
Icons problems is not from appmenu. Why are you think than appmenu affects icons?

**Comment by darcn181** [comment]
I've finally found out what was causing the issue(i think), it seems that it was all the time the xfce-gtk3 packages from manjaro. I thought it was the applet because after compiling it in the new xfce the problems started happening. I uninstalled the appmenu and the problem is still in the pc so i think it was all the time xfce because i uninstalled the gtk3 version and reinstalled the stable version and the problem was gone.
